MyApplication/app/src/main/java/com/joelimyx/myapplication/MainActivity.java

```diff
@@ -0,0 +1,114 @@
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+    RecyclerView mRecyclerView;
+    @BindView(R.id.button1) Button mButton1;
+    Button mButton2;
+    @BindView(R.id.button3) Button mButton3;
+    public static final StringBuilder GET_URL_BUILDER = new StringBuilder("http://api.walmartlabs.com/v1/search?format=json&apiKey=xp8nz8badbdjrn2pb9r7bsub&query=");
+    private MainAdapter mAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mRecyclerView = (RecyclerView) findViewById(R.id.recyclerview);
+        mButton1 = (Button) findViewById(R.id.button1);
+        mButton2 = (Button) findViewById(R.id.button2);
+        mButton3 = (Button) findViewById(R.id.button3);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false));
+
+        ConnectivityManager manager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+        final NetworkInfo networkInfo = manager.getActiveNetworkInfo();
+        View.OnClickListener listener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+            if (networkInfo!=null && networkInfo.isConnected()){
+                String temp="";
+                switch (view.getId()){
+                    case R.id.button1:
+                        temp = "cereal";
+                        break;
+                    case R.id.button2:
+                        temp = "tea";
+                        break;
+                    case R.id.button3:
+                        temp = "chocolate";
+                        break;
+                }
+                getResponse(GET_URL_BUILDER.toString()+temp);
+            }else {
+                Toast.makeText(MainActivity.this, "No Connection", Toast.LENGTH_SHORT).show();
+            }
+            }
+        };
+        mButton1.setOnClickListener(listener);
+        mButton2.setOnClickListener(listener);
+        mButton3.setOnClickListener(listener);
+    }
+    private void getResponse(String url){
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder().url(url).build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()){
+                    throw new IOException("BAD GET");
```

Comment:
You'll want to notify the user here too, otherwise they'll think your app just froze and didn't do anything